src/modules/paymentProcessor/StreamingPaymentProcessor.sol

```diff
@@ -744,8 +767,14 @@ contract StreamingPaymentProcessor is Module, IStreamingPaymentProcessor {
 
     /// @notice validate uint start input.
     /// @param _start uint to validate.
+    /// @param _cliff uint to validate.
+    /// @param _end uint to validate.
     /// @return True if uint is valid.
-    function validStart(uint _start) internal view returns (bool) {
-        return !(_start < block.timestamp || _start >= type(uint).max);
+    function validTimes(uint _start, uint _cliff, uint _end)
+        internal
+        view
+        returns (bool)
+    {
+        return !(_start >= type(uint).max && _start + _cliff > _end);
```

Comment:
I couldn't see a reason why (a) streaming payments couldn't start in the past (so far they needed to be `block.timestamp`or greater) and (b) streaming payments were allowed to have an end date that was smaller than the start date (I can see a reason for `start == end` for instant payments via the streaming processor, but not for `end < start`.

I adapted the implementation so that `start + cliff <= end` - with `start` not needing to be greater than or equal to `block.timestamp`. I can at least see good reason for vesting schemes to start vesting retroactively, especially with cliffs.

Is there anything I am missing @FHieser on why this was implemented this way? Any reason why my change shouldn't be done?

There was this `testProcessPaymentsWorksForEndTimeThatIsPlacedBeforeStartTime` which I found and why I came to this question. This test seems to work with the new implementation but I assume that is not how it was intended. I will continue looking into this tomorrow :D Any input on why things have been done this way would be great :)